src/snowflake/cli/plugins/object/stage/diff.py

```diff
@@ -154,7 +161,36 @@ def build_md5_map(list_stage_cursor: SnowflakeCursor) -> Dict[str, str]:
     }
 
 
-def stage_diff(local_path: Path, stage_fqn: str) -> DiffResult:
+def _get_relative_paths_to_sync(
+    relative_files_to_sync: List[Path], local_path: Path, remote_paths: Set[str]
+) -> List[str]:
+    paths = []
+    for file in relative_files_to_sync:
+        path = Path(os.path.join(local_path, file))
+        relpath = path.relative_to(local_path)
+        if not path.exists() and str(relpath) not in remote_paths:
+            raise FileError(
+                str(file), "This file does not exist either locally or remotely"
+            )
+        elif path.is_dir():
+            raise ClickException(f"Specifying directories is not supported: '{file}'")
+        else:
+            paths.append(str(relpath))
+    return paths
+
+
+def _filter_from_diff(result: DiffResult, paths_to_keep: Set[str]) -> DiffResult:
```

Comment:
From https://github.com/snowflakedb/snowflake-cli/pull/933#discussion_r1539911756, it sounded like we want to keep `stage_diff` untouched, and filter only once the entire result has been returned from the function. This filtering would happen in `sync_deploy_root_with_stage`. Are we deviating from that understanding?